src/MongoDB.Driver.Core/Core/Servers/ServerDescription.cs

```diff
@@ -437,7 +438,7 @@ public bool Equals(ServerDescription other)
                 object.Equals(_canonicalEndPoint, other._canonicalEndPoint) &&
                 object.Equals(_electionId, other._electionId) &&
                 EndPointHelper.Equals(_endPoint, other._endPoint) &&
-                object.Equals(_heartbeatException, other._heartbeatException) &&
+                ExceptionHelper.Equals(_heartbeatException, other._heartbeatException) &&
```

Comment:
This is a general purpose Equals method and should use the standard Equals method when comparing exceptions, so DON'T change this line.